### DISCUSSION TITLE
Make autoclose Merge-Requests for unreachable packages optional

### DISCUSSION THREAD

**Discussion by MPritsch** [renovatebot/renovate#25371] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.6.1

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

We have some privately hosted repositories/registries mirrors which can be unreliable at times. Since Renovate can't reach the registry at it's scheduled run, it autocloses the corrersponding Merge-Request. When it later runs again, it creates a new Merge-Request.

This means that during the time the registry was not available all our Merge-Requests for that particular package-type (e.g. npm) will be closed in all our gitlab-repositories. This is confusing for our developers since this usually means that all packages are up to date.

I understand that for public registries an failed request usually means, the package is no longer available. But in our case the whole registry-server does not respond.

I would love some configuration to allow Merge-Requests to stay open even for failed look up requests.

### What you tried so far.

Read renovate documentation, config presets and discussions here on github.

Not sure how you would differentiate this from an usual "not found" response. We receive an "ERRCONNREFUSED" according to the renovate debug-logs.

If you need specific logs, I can provide them. But at this stage I just require general assistance.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by MPritsch** [comment]
Bump, we still have this issue. Any advice?

**Comment by rarkins** [comment, marked as answer]
You can configure pruneStaleBranches=false to disable autoclosing of any branch permanently. If you want more advanced, dynamic behavior, such as "disable prune branches whenever there's a lookup error" then it would require a feature request.
